pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartialUpsertHandler.java

```diff
@@ -33,18 +34,25 @@
 public class PartialUpsertHandler {
   // _column2Mergers maintains the mapping of merge strategies per columns.
   private final Map<String, PartialUpsertMerger> _column2Mergers = new HashMap<>();
+  private final UpsertConfig.Strategy _defaultPartialUpsertStrategy;
```

Comment:
Store a default merger (`PartialUpsertMergerFactory.getMerger(defaultPartialUpsertStrategy)`) instead of the strategy to avoid per record creation of the merger

Reply:
`PartialUpsertMergerFactory.getMerger` actually returns the same singleton instance. So, there is only one merger instance created. 
But I can make this change as it get rids of the assumption in the underlying implementation.